### PR TITLE
Schema: Incorporate Cue research

### PR DESCRIPTION
**Pull request by matthewmueller**
[Rendered View](https://github.com/prisma/specs/blob/prisma-schema-cue/prisma-schema/Readme.md)

**Added**

- How to integrate the lower-level type definitions brought by connectors.
- Possible syntax for providing arguments to types
- How to merge model's datatype together

### CODE REVIEW

File: prisma-schema/Readme.md [hunk 1]
```diff
@@ -814,17 +815,64 @@ model Customer {}
...
 You can attach any field attribute to a type definition.
 
+### Type Definitions provided by Connectors
+
+Connectors can bring their own type definitions allowing you to use these types in your own schemas.
+
+**postgres.prisma (generated by the connector)**
+
+```groovy
+type SmallInt Int @raw("smallint") @bound(gte: -32768, lte: 32767)
```
Comment (matthewmueller) [inline prisma-schema/Readme.md]: FYI, I changed `@constraint` to `@bound` for no other reason than bound is shorter

**Comment by pantharshit00** [comment]
Bulk creation and updation

https://github.com/prisma/prisma2/issues/284

**Comment by matthewmueller** [comment]
Sorry, I tried merging this awhile back but ran into "rebase hell". Will try again asap

**Comment by matthewmueller** [comment]
merged spec

**Pull request merged**